### DISCUSSION TITLE
Event handlers called multiple times

### DISCUSSION THREAD

**Discussion by aalmada** [ethereum-ts/TypeChain#689] [Q&A] [answered]
Hi!
I've noticed that when an event is emitted, the handler is called more than once. It's actually called using data from previous events.
For example, a handler for the `Transfer(null, address)` filter, when the `tokenId` 20 is minted, the event handler is called in burst for the ids 15, 16, 17, 18, 19, and 20.
I don't know if this is expected or an issue with Typechain or ethers. 
If expected, how do you usually handle it?

**Comment by krzkaczor** [comment, marked as answer]
It's an ethers feature (maybe releated to reorgs). TypeChain does not generate runtime code.

**Reply by aalmada** [reply]
Thank you! I found this discussion on "listening for future events": https://github.com/ethers-io/ethers.js/discussions/1523
